vm/vm.go

```diff
@@ -156,6 +175,9 @@ func (vm *VirtualMachine) Run(ctx context.Context) (err error) {
 	vm.activateCode(0, vm.ip, code)
 	ctx = object.WithCallFunc(ctx, vm.callFunction)
 	ctx = limits.WithLimits(ctx, vm.limits)
+	if vm.concAllowed {
+		ctx = object.WithSpawnFunc(ctx, vm.spawnFunction)
+	}
 	err = vm.eval(ctx)
 	return
```

Comment:
I think Risor should wait for dangling threads/goroutines.

Currently, the following code prints nothing:

```go
go print("hello")
```

It spawns the thread, but doesn't get to executing it before the application ends.

This does have parity with Go, which also just stops without considering dangling goroutines. Such as the following Go code also prints nothing when executed:

```go
package main

import "fmt"

func main() {
	go fmt.Println("hello")
}
```

But I think this is a annoying "gotcha" to be careful of. I'd be best if the user always waits for their goroutines to finish, but some support from the runtime when you just forget would be nice.

If you think it's just "bad risor code" to forget waiting, then I suggest to at least add a warning message when Risor's execution ends, that says something like "WARN: risor execution ends while still having unfinished goroutines". Or maybe it logs an error AND waits for the goroutines?

Reply:
Hmm. That would be a nice capability to have, but I'm not sure it's necessary to have right now. It could be added later as an option. And having Risor match Go's behavior exactly makes some sense too.

Having a centralized registry of running Risor threads is going to be a source of complexity and potentially lock contention.

Reply:
> And having Risor match Go's behavior exactly makes some sense too.

Unless the Go's behavior is bad

Reply:
I think waiting should be done explicitly, maybe Ido wnat to have some threads that run till my program finishes, implicit waiting could be limiting and lead to issues.